File: MiniDOMBase/DMD/Luoghi/Regioni/CRegioniCursor.cs
using System;
using System.Collections;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using static minidom.Sistema;
using static minidom.Anagrafica;
using minidom.repositories;
using System.Collections.Generic;


namespace minidom
{
    public partial class Anagrafica
    {

        /// <summary>
        /// Cursore sulla tabella delle regioni
        /// </summary>
        [Serializable]
        public class CRegioniCursor 
            : minidom.Databases.DBObjectCursorPO<CRegione>
        {
            private DBCursorStringField m_Sigla = new DBCursorStringField("Sigla");
            private DBCursorStringField m_Nazione = new DBCursorStringField("Nazione");
            
            /// <summary>
            /// Costruttore
            /// </summary>
            public CRegioniCursor()
            {
            }

            
            /// <summary>
            /// Sigla
            /// </summary>
            public DBCursorStringField Sigla
            {
                get
                {
                    return m_Sigla;
                }
            }

            /// <summary>
            /// Nazione
            /// </summary>
            public DBCursorStringField Nazione
            {
                get
                {
                    return m_Nazione;
                }
            }
 
            //public override object InstantiateNew(DBReader dbRis)
            //{
            //    return new CRegione();
            //}

            //public override string GetTableName()
            //{
            //    return "tbl_Luoghi_Regioni";
            //}


            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Anagrafica.Luoghi.Regioni; //.Module;
            }

            //protected internal override CDBConnection GetConnection()
            //{
            //    return Luoghi.Database;
            //}
        }
    }
}

File: MiniDOMBase/DMD/Luoghi/Regioni/CRegioniCursor.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases
Imports minidom.Anagrafica

Partial Public Class Anagrafica

    Public Class CRegioniCursor
        Inherits DBObjectCursorPO(Of CRegione)

        Private m_Nome As New CCursorFieldObj(Of String)("Nome")
        Private m_NumeroAbitanti As New CCursorField(Of Integer)("NumeroAbitanti")
        Private m_Sigla As New CCursorFieldObj(Of String)("Sigla")
        Private m_Nazione As New CCursorFieldObj(Of String)("Nazione")
        Private m_NomeAbitanti As New CCursorFieldObj(Of String)("NomeAbitanti")

        Public Sub New()
        End Sub

        Public ReadOnly Property Nome As CCursorFieldObj(Of String)
            Get
                Return Me.m_Nome
            End Get
        End Property

        Public ReadOnly Property NumeroAbitanti As CCursorField(Of Integer)
            Get
                Return Me.m_NumeroAbitanti
            End Get
        End Property

        Public ReadOnly Property Sigla As CCursorFieldObj(Of String)
            Get
                Return Me.m_Sigla
            End Get
        End Property

        Public ReadOnly Property Nazione As CCursorFieldObj(Of String)
            Get
                Return Me.m_Nazione
            End Get
        End Property

        Public ReadOnly Property NomeAbitanti As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeAbitanti
            End Get
        End Property

        Public Overrides Function InstantiateNew(dbRis As DBReader) As Object
            Return New CRegione
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_Luoghi_Regioni"
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Luoghi.Regioni.Module
        End Function

        Protected Friend Overrides Function GetConnection() As Databases.CDBConnection
            Return Anagrafica.Luoghi.Database
        End Function
    End Class

End Class